docs/learn.md

```diff
@@ -126,10 +126,281 @@ OCaml. Some examples are the upgrade of the compiler version used by Melange to
+to be done by hand. It is as simple as adding the name of the package in the
+`depends` field.
+
+#### Linking packages for development
+
+This can be achieved with `opam pin`. For example, to pin a package to a
+specific commit on GitHub:
+
+```bash
+opam pin add reason-react.dev https://github.com/reasonml/reason-react.git#61bfbfaf8c971dec5152bce7e528d30552c70bc5
+```
+
+Branch names can also be used.
+
+```bash
+opam pin add reason-react.dev https://github.com/reasonml/reason-react.git#feature
+```
+
+A shortcut to get the latest version of a package is to use the `--dev-repo`
+flag, e.g.
+
+```bash
+opam pin add reason-react.dev --dev-repo
+```
+
+To remove the pinning for any package, use `opam unpin <package_name>` or `opam
+pin remove <package_name>`.
+
+For other options, the command is well described in [the official
+documentation](https://opam.ocaml.org/doc/Usage.html#opam-pin).
+
+#### Upgrading packages
+
+To upgrade the installed packages to the latest version, run:
+
+```bash
+opam upgrade <package_name>
+```
+
+`opam upgrade` is also able to upgrade *all* the packages of the local switch if
+no package name is given.
+
+There is one big difference compared to npm: opam stores a local copy of the
+opam repository, like `apt-get` does in Debian. So we often want to update this
+copy before requesting an upgrade:
+
+```bash
+opam update && opam upgrade <package_name>
+```
+
+#### Dev dependencies
+
+You can use the [`with-dev-setup`
+field](https://opam.ocaml.org/doc/Manual.html#pkgvar-with-dev-setup) to define
+dependencies that are only required at development time. For example:
+
+```
+depends: [
+  "ocamlformat" {with-dev-setup}
+]
+```
+
+This has to be combined with the `--with-dev-setup` flag when installing
+dependencies, e.g. `opam install --deps-only --with-dev-setup`.
+
+#### Lock files
+
+Lock files are also not common yet in the opam world, but they can be used as
+follows:
+
+- Using `opam lock` to generate the lock file when needed (basically after each
+  `opam install` or `opam upgrade`).
+- Adding `--locked` to all the `opam install --deps-only` and `opam switch
+  create .` commands.
+
+
+#### Bindings and package management
+
+When writing Melange libraries that bind to existing JavaScript packages, the
+users of the Melange library will have to make sure to install those JavaScript
+packages manually.
+
+This is similar to how OCaml bindings to system libraries work, see examples
+like
+[`ocaml-mariadb`](https://github.com/andrenth/ocaml-mariadb/blob/9db2e4d8dec7c584213d0e0f03d079a36a35d9d5/README.md?plain=1#L18-L20)
+or
+[`ocurl`](https://github.com/ygrek/ocurl/blob/f0c6f47d6f3d25282648439dc4ade5810a993710/README.md?plain=1#L16).
+
+The advantage of this approach —as opposed to vendoring the JavaScript packages
+inside the bindings— is that it gives users of the bindings complete flexibility
+over the way these JavaScript packages are downloaded and bundled.
```

Comment:
it's also a bit of a limitation that we'd like to lift in the future.

e.g. melange should be able to specify that reason-react works with `react` and `react-dom` versions < x >= y

Reply:
Yeah, I think this could be the place where we mention something like

> In the future, we might explore improving this integration by using an approach similar to `depext`

But I am not very comfortable mentioning potential features in the manual that are not in the roadmap yet. 🤔